photon-lib/src/main/scala/com/linkedin/photon/ml/data/ScoredGameDatum.scala

```diff
@@ -0,0 +1,146 @@
+   * [[weight]] and [[idTypeToValueMap]] are identical. However, for performance reasons, we do not assert this
+   * requirement and arbitrarily use the current instance's values for these variables in the returned datum
+   *
+   * The score of the returned datum is combined using the provided operator
+   *
+   * @param op The operator to combine the scores of two data
+   * @param that [[ScoredGameDatum]] to be combined
+   * @return Combined scored datum
+   */
+  def combineScores(op: (Double, Double) => Double, that: ScoredGameDatum): ScoredGameDatum = {
+    ScoredGameDatum(this.response, this.offset, this.weight, op(this.score, that.score), this.idTypeToValueMap)
+  }
+
+  /**
+   * The plus operation for scored data. Two data can only be added if their [[response]], [[offset]], [[weight]] and
+   * [[idTypeToValueMap]] are identical. See [[similarTo()]]
+   *
+   * @param that The other scored datum instance
+   * @return A new scored datum instance with score being the sum of the two input scores
+   */
+  def +(that: ScoredGameDatum): ScoredGameDatum = {
+    this.combineScores((a, b) => a + b, that)
+  }
+
+  /**
+   * The subtract operation for scored data. Two data can only be subtracted if their [[response]], [[offset]],
+   * [[weight]] and [[idTypeToValueMap]] are identical. See [[similarTo()]]
+   *
+   * @param that The other scored datum instance
+   * @return A new scored datum instance with score being the subtraction of the two input scores
+   */
+  def -(that: ScoredGameDatum): ScoredGameDatum = {
+    this.combineScores((a, b) => a - b, that)
+  }
+
+  /**
+   * Check if two scored data are equal. Two scored data are equal if all their fields are identical
+   *
+   * @param that the other scored datum instance
+   * @return whether current instance is equal to the input datum
+   */
+  override def equals(that: Any): Boolean = {
+    that match {
+      case other: ScoredGameDatum => similarTo(other) && isAlmostZero(this.score - other.score)
+      case _ => false
+    }
+  }
+
+  /**
+   * @return string representation of the scored datum
+   */
+  override def toString: String = {
+    s"[response=$response, offset=$offset, weight=$weight, score=$score, idTypeToValueMap=$idTypeToValueMap]"
+  }
+}
+
+object ScoredGameDatum {
+  val ZERO_SCORE = 0.0
+
+  def apply(
+    labeledPoint: LabeledPoint,
+    score: Double,
+    idTypeToValueMap: Map[String, String] = Map()): ScoredGameDatum = {
+    new ScoredGameDatum(labeledPoint.label, labeledPoint.offset, labeledPoint.offset, score, idTypeToValueMap)
+  }
+
+  /**
+   * Get a default scored datum instance
+   *
+   * @param score input score (default: [[ZERO_SCORE]])
+   * @return Returns a default scored datum with a response=1.0, offset=0.0, weight=1.0, score=input score or default
+   *         and an empty map
+   */
+  def getDefaultScoredDatumWithScore(score: Double = ZERO_SCORE): ScoredGameDatum = {
```

Comment:
How about using default values on `apply` to avoid a lengthy function name? Or a second `apply` which allows only setting the score.